### ISSUE TITLE
Not redirected properly (ingress-nginx)

### ISSUE THREAD

**Issue by Djiit** [oauth2-proxy/oauth2-proxy#921]
## Expected Behavior

I expect oauth2-proxy to redirect to app.domain.com/api/foo (to honor the redirect url)

## Current Behavior

After the login process, I'm redirected to oauth2-proxy.domain.com (my 

## Possible Solution

N/A

## Steps to Reproduce (for bugs)

I'm using a vanilla ingress-nginx ingress controller. oauth-2proxy is hosted at its own subdomain, not the same as the apps.

My app is using this Ingress : 
```yml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    nginx.ingress.kubernetes.io/auth-signin: https://oauth2-proxy.domain.com/oauth2/start?rd=$scheme%3A%2F%2F$host$escaped_request_uri
    nginx.ingress.kubernetes.io/auth-url: https://oauth2-proxy.domain.comoauth2/auth
```

And this is the env of my oauth2-proxy : 

```
extraEnv:
  - name: OAUTH2_PROXY_COOKIE_DOMAINS
    value: ".domain.com"
  - name: OAUTH2_PROXY_WHITELIST_DOMAINS
    value: ".domain.com"
  - name: OAUTH2_PROXY_EMAIL_DOMAIN
    value: "*"
  - name: OAUTH2_PROXY_SKIP_JWT_BEARER_TOKENS
    value: "true"
  - name: OAUTH2_PROXY_SKIP_PROVIDER_BUTTON
    value: "true"
  - name: OAUTH2_PROXY_PASS
    value: "true"
  - name: OAUTH2_PROXY_PROVIDER
    value: "oidc"
  - name: OAUTH2_PROXY_PROVIDER_DISPLAY_NAME
    value: "Auth0"
  - name: OAUTH2_PROXY_INSECURE_OIDC_ALLOW_UNVERIFIED_EMAIL
    value: "true"
  - name: OAUTH2_PROXY_OIDC_ISSUER_URL
    value: "https://domain.auth0.com/"
  - name: OAUTH2_PROXY_REDIRECT_URL
    value: "https://oauth2-proxy.domain.com/oauth2/callback"
```

Please note some parts where replace by `domain.com` on purpose.

## Your Environment

- Version used: 6.1.1
- ingress-nginx controller.

Seems similar to https://github.com/oauth2-proxy/oauth2-proxy/issues/782

**Comment by JoelSpeed** [comment]
Is there anything noted in the logs about why the redirect was rejected? I believe in 6.1.1 it should log a reason as to why it was a redirect was not honoured

**Comment by Djiit** [comment]
Oh man I was using `OAUTH2_PROXY_WHITELIST_DOMAIN` without a `s`. This is working now.
In the docs, it's referred as `whitelist-domain` (singular form) should we update anything?

**Comment by JoelSpeed** [comment]
There's a note in the configuration doc that explains when the option should be singular vs plural already, this is a common trap though, need to make it more obvious somehow :/

**Comment by Djiit** [comment]
nah, thanks to this note I tried with a `s`, so, I guess it worked :). Still blocked on JWT validation with Auth0. I'm going to open a separate issue. Thanks
